Description: Backport of status-request-revoked.c from groovy
Origin: upstream
Bug: https://gitlab.com/gnutls/gnutls/-/issues/967
Forwarded: not-needed
Applied-Upstream: https://gitlab.com/gnutls/gnutls/-/commit/ed208fe55f31478732fd6cc394f9576b315a42cd
Last-Update: 2021-03-04
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: gnutls28-3.6.13/tests/status-request-revoked.c
===================================================================
--- gnutls28-3.6.13.orig/tests/status-request-revoked.c
+++ gnutls28-3.6.13/tests/status-request-revoked.c
@@ -65,6 +65,16 @@ static void client_log_func(int level, c
 	fprintf(stderr, "client|<%d>| %s", level, str);
 }
 
+static time_t mytime(time_t * t)
+{
+	time_t then = 1586000000;
+
+	if (t)
+		*t = then;
+
+	return then;
+}
+
 static unsigned char server_cert_pem[] =
 "-----BEGIN CERTIFICATE-----\n"
 "MIIEKjCCAhKgAwIBAgIIRiBQA6KFBj0wDQYJKoZIhvcNAQELBQAwDzENMAsGA1UE\n"
@@ -252,6 +262,7 @@ static int cert_verify_callback(gnutls_s
 	unsigned int status;
 	int ret;
 
+	gnutls_global_set_time_function(mytime);
 	ret = gnutls_certificate_verify_peers2(session, &status);
 	if (ret < 0)
 		return -1;
